pkg/errors/error.go

```diff
@@ -37,6 +37,15 @@ func NewForbiddenError(message, details string) *Error {
 	}
 }
 
+func NewUnauthorizedError(message, details string) *Error {
+	return &Error{
+		Status:  http.StatusText(http.StatusUnauthorized),
+		Code:    http.StatusUnauthorized,
+		Message: message,
+		Details: details,
+	}
+}
```

Comment:
But how it would help us? We have to return an http error response (`http.Error()`) which requires three params: the http response writer, the status code (401 in our case) and the error message. We provide all of it.
`Reason: "Unauthorized"` is an implementation detail in the kube status error object which doesn't add anything to the http error we are already returning. Unless I'm missing something.

Reply:
The main motivation was to return the same error as K8s would return, so the client doesn't recognize that it's not a real K8s server. If any client would use the `Reason` for anything (as it's always returned by K8s) then I thought that it would be fine to add it there.
But I agree that it's just a detail and that `Status`, `Code`, and `Message` are the only fields that really matter.

Reply:
OK. I'm guessing that you are talking about the http response body here. Basically the JSON returned by the kube or proxy.
For now proxy returns `err.Error()` as the error body. It doesn't marshal any JSON. So the underlying error struct doesn't really matter.
We need to improve it and return the same JSON a kube server would return but it requires more work than just changing our internal error structs (or switching to kube error objects).
So it's a valid point but let's do it in a separate PR. I will add it to the TODO list in the PR description.